### PR TITLE
Release v4.5.0

### PR DESCRIPTION
**Pull request by joyet-simon**
## Reason for change

[Linear task](https://linear.app/almapay/issue/MPP-xxx/)

### Code changes

Release withe deferred capture and pre-commit

### How to test

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: cartridges/int_alma/cartridge/scripts/helpers/almaPaymentHelper.js [hunk 1]
```diff
@@ -281,25 +300,73 @@ function createPayment(param) {
...
+
+/**
+ * Check if manual capture is available
+ * @param {boolean} isManualCaptureEnabled  manual capture is enabled
+ * @param {number} installmentsCount installments count
+ * @param {number} deferredDays deferred days
+ * @return {boolean} is available
+ */
+function isAvailableForManualCapture(isManualCaptureEnabled, installmentsCount, deferredDays) {
+    return isManualCaptureEnabled && installmentsCount <= 4 && deferredDays <= 0;
```
Comment (joyet-simon) [inline cartridges/int_alma/cartridge/scripts/helpers/almaPaymentHelper.js:346]: Add constant.

File: cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js [hunk 2]
```diff
@@ -186,6 +185,16 @@ function getFormattedPlans(plans) {
...
         Object.keys(plans[paymentMethod]).forEach(function (keys) {
             paymentMethodPlans.push(plans[paymentMethod][keys]);
+            plans[paymentMethod][keys].captureMethod = 'automatic';
+
+            if (almaPaymentHelper.isAvailableForManualCapture(
+                almaConfigHelper.isDeferredCaptureEnable(),
+                plans[paymentMethod][keys].installments_count,
+                plans[paymentMethod][keys].deferred_days
+            )) {
+                plans[paymentMethod][keys].captureMethod = 'manual';
```
Comment (joyet-simon) [inline cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js:195]: Add constant.

File: cartridges/int_alma/cartridge/scripts/steps/CapturePaymentOrders.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+var OrderMgr = require('dw/order/OrderMgr');
+var Order = require('dw/order/Order');
+var Status = require('dw/system/Status');
+var almaOrderHelper = require('*/cartridge/scripts/helpers/almaOrderHelper');
+var almaPaymentHelper = require('*/cartridge/scripts/helpers/almaPaymentHelper');
+var Logger = require('dw/system/Logger');
+
+exports.execute = function () {
+    var Capture = almaPaymentHelper.Capture;
```
Comment (joyet-simon) [inline cartridges/int_alma/cartridge/scripts/steps/CapturePaymentOrders.js:11]: uppercase

File: cartridges/int_alma/cartridge/scripts/steps/CapturePaymentOrders.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    var orders = OrderMgr.searchOrders(
+        "custom.ALMA_Deferred_Capture_Status='ToCapture' and status != {0} and status != {1}",
```
Comment (joyet-simon) [inline cartridges/int_alma/cartridge/scripts/steps/CapturePaymentOrders.js:13]: const

File: cartridges/int_alma/cartridge/scripts/steps/CapturePaymentOrders.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        null,
+        Order.ORDER_STATUS_FAILED,
+        Order.ORDER_STATUS_CANCELLED
+    );
+
+    if (orders.count > 0) {
+        while (orders.hasNext()) {
```
Comment (joyet-simon) [inline cartridges/int_alma/cartridge/scripts/steps/CapturePaymentOrders.js:20]: extract function

File: cartridges/int_alma/cartridge/scripts/steps/CheckRefund.js [hunk 6]
```diff
@@ -42,27 +41,50 @@ function refundPaymentForOrder(order) {
...
 
     Logger.info('[INFO][ALMA refund] job launched for: ' + orders.count + ' orders.');
     if (orders.count > 0) {
         while (orders.hasNext()) {
             var orderItem = orders.next();
             if (isOrderToBeRefund(orderItem)) {
                 try {
-                    refundPaymentForOrder(orderItem);
+                    if (orderItem.custom.ALMA_Deferred_Capture_Status === 'ToCapture') {
+                        var amount = 0;
```
Comment (joyet-simon) [inline cartridges/int_alma/cartridge/scripts/steps/CheckRefund.js:55]: extract function

File: cartridges/int_alma/cartridge/static/default/js/almaCheckout.js [hunk 7]
```diff
@@ -179,11 +180,13 @@ window.addEventListener('DOMContentLoaded',
                 var deferred_days = parseInt(t.getAttribute('data-deferred-days'));
                 var alma_payment_method = t.getAttribute('data-alma-payment-method');
                 var in_page = t.getAttribute('data-in-page') === 'true';
+                var captureMethod = t.getAttribute('data-captureMethod');
 
                 if (in_page) {
                     await inPageInitialize(
                         t.id + "-inpage",
```
Comment (joyet-simon) [inline cartridges/int_alma/cartridge/static/default/js/almaCheckout.js:187]: rename t

**Review by Francois-Gomis — APPROVED** [review]

**Pull request merged**